src/components/data_comps/datm/nuopc/datm_comp_nuopc.F90

```diff
@@ -534,7 +549,7 @@ subroutine InitializeRealize(gcomp, importState, exportState, clock, rc)
     ! Set the coupling scalars
     !--------------------------------
 
-    call shr_nuopc_dmodel_AvectToState(d2x, exportState, grid_option, rc=rc)
+    call shr_nuopc_grid_ArrayToState(d2x%rattr, seq_flds_a2x_fields, exportState, grid_option, rc=rc)
```

Comment:
Why does this interface need the field list (seq_flds_a2x_fields) argument?

Reply:
If you look at the interface in shr_nuopc_grid_ArrayToState you will see that  this argument is passed in as rList and gets used as follows:
   
call shr_string_listGetName(rList, nf, fldname, dbrc)
call ESMF_StateGet(state, itemName=trim(fldname), field=lfield, rc=rc)